### ISSUE TITLE
How to debug pitest 

### ISSUE THREAD

**Issue by snowe2010** [szpak/gradle-pitest-plugin#111]
Adding 

```kotlin
    mainProcessJvmArgs = listOf("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005")
```
does not seem to be enough. As soon as I connect, none of my debug points are hit. Do you have any sort of instructions for debugging the pitest run (in this gradle plugin) using IntelliJ?

**Comment by szpak** [comment]
I believe it worked in the past. I will try to reproduce the problem.

**Comment by szpak** [comment]
I cannot reproduce that issue. Adding the aforementioned parameters to `mainProcessJvmArgs` (in Gradle configuration in Groovy) I was able to convince PIT execution to wait for a debugger and later on the execution stopped on my breakpoint in the main method in `MutationCoverageReport`.

Can you re-ckeck your setup?

**Comment by snowe2010** [comment]
I guess I'm just unsure of how to tell the gradle plugin to use my local copy of pit for the debug points. I have managed to get the gradle plugin to wait for my debugger to connect, but no debug points are hit.

**Comment by szpak** [comment]
I checked it with PIT imported into my IDE.

What do you mean by "to get the gradle plugin to wait for my debugger to connect"? The Gradle plugin should execute as is. PIT is executed as a separate process and there you should attach a debugger (after my plugin starts PIT it should hang waiting for debugger).

**Comment by snowe2010** [comment]
sorry yes that's what I mean. I add the above flags, wait for the gradle plugin to kick off the PIT process, which then waits for a debugger to be attached, attach the debugger, but then everything continues without hitting the breakpoints. I guess I will just try again.

**Comment by szpak** [comment]
Any luck with your debugging?

**Comment by snowe2010** [comment]
Nope. Well sorta. I am able to get a debug point to hit in `MutationCoverageReport`, but I don't need it to stop there. I need it to stop in `CoverageMinion`, but it appears it only stops in this one module.

**Comment by snowe2010** [comment]
It seems this might be because it calls out a second Process, but if I add 
`jvmArgs = listOf("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005")`

then I instead get 
```
7:36:09 PM PIT >> INFO : MINION : ERROR: JDWP Non-server transport dt_socket must have a connection address specified through the 'address=' option
ERROR: JDWP invalid option: -agentlib:jdwp=transport=dt_socket
```

**Comment by szpak** [comment]
That's because PIT itself starts minions using a separate JVM instances :).

**Comment by szpak** [comment]
@hcoles I haven't found in the project FAQ how to debug minions. Is it possible to do that in any easy way?

**Comment by szpak** [comment]
Investigating https://github.com/hcoles/pitest/issues/822 it turned out that debugging minions is definitely possible, but I needed some extra hacks to achieve that (for the reasons you encountered).

I reported https://github.com/jopt-simple/jopt-simple/issues/147 to check if my problem with passing extra arguments can be resolved at the parser level. IF not, some extra changes in PIT would need to be performed.
